caddy_test.go

```diff
@@ -56,3 +59,38 @@ func TestIsLoopback(t *testing.T) {
 		}
 	}
 }
+
+func TestListenerAddrEqual(t *testing.T) {
+	ln1, err := newLocalListener("[::]:2016")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ln1.Close()
+
+	ln2, err := newLocalListener("[::]:2017")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ln2.Close()
+
+	for i, test := range []struct {
+		ln     net.Listener
+		addr   string
+		expect bool
+	}{
+		{ln1, "0.0.0.0:2016", true},
+		{ln2, "0.0.0.0:2018", false},
```

Comment:
Can we add some more test cases here? Use the test coverage tool to ensure all lines of the `listenerAddrEqual()` function are being hit.